routes/stats.py

```diff
@@ -175,48 +171,196 @@ def stats() -> Union[Response, str]:
+
+            # Generate date label
+            dfmtstr = "%a %-d. %b"
+            labels.append(start.strftime(dfmtstr))
+
+            # Get query count for day
+            q = list(QueryCountQuery.period(start, end, enclosing_session=session))
+            query_count_data.append(q[0][0])
+
+    query_avg = sum(query_count_data) / num_days
+
+    start = today - timedelta(days=num_days)
+    end = datetime.utcnow()
+
+    # Query types
+    res = QueryTypesQuery.period(
+        start=start,
+        end=end,
+        enclosing_session=session,
+    )
+
+    # This function is used to ensure that all the query
+    # types have a fixed, unique color on the pie chart.
+    def _hexcolor4string(s: str) -> str:
+        """Generate a hex color from a string."""
+        hash = md5(s.encode("utf-8")).hexdigest()
+        hash_values = (
+            hash[:8],
+            hash[8:16],
+            hash[16:24],
+        )
+        rgb = tuple(int(value, 16) % 256 for value in hash_values)
+        return "#%02x%02x%02x" % rgb
+
+    query_types_data = {
+        "labels": [k[1] for k in res],
+        "datasets": [
+            {
+                "data": [k[0] for k in res],
+                "backgroundColor": [_hexcolor4string(k[1]) for k in res],
+            }
+        ],
+    }
+
+    # Client types
+    _CLIENT_COLORS = {
+        "ios": "#4c8bf5",
+        "ios_flutter": "#4c8bf5",
+        "android": "#a4c639",
+        "android_flutter": "#a4c639",
+        "python": "#ffff00",
+        "www": "#f7b924",
+    }
+    res = QueryClientTypeQuery.period(start, end)
+    client_types_data = {
+        "labels": [f"{k[0]} {k[1] or ''}".rstrip() for k in res],
+        "datasets": [
+            {
+                "data": [k[2] for k in res],
+                "backgroundColor": [_CLIENT_COLORS.get(k[0], "#ccc") for k in res],
+            }
+        ],
+    }
+
+    # Top queries (answered and unanswered)
+    def prep_top_answ_data(res) -> List[Dict[str, Any]]:
+        rl = list(res)  # Consume generator
+        highest_count = rl[0][2]
+        toplist = []
+        for q in rl:
+            toplist.append({"query": q[0], "count": q[2], "freq": q[2] / highest_count})
+        return toplist
+
+    res = TopUnansweredQueriesQuery.period(start, end, enclosing_session=session)
+    top_unanswered = prep_top_answ_data(res)
+    res = TopAnsweredQueriesQuery.period(start, end, enclosing_session=session)
+    top_answered = prep_top_answ_data(res)
+
+    return {
+        "query_count": {
+            "labels": labels,
+            "datasets": [{"data": query_count_data}],
+            "avg": query_avg,
+        },
+        "query_types": query_types_data,
+        "client_types": client_types_data,
+        "top_unanswered": top_unanswered,
+        "top_answered": top_answered,
+    }
+
+
+@routes.route("/stats/queries", methods=["GET"])
+@cache.cached(timeout=30 * 60, key_prefix="stats", query_string=True)
```

Comment:
Do you need both `@cache.cached` and `@max_age`?

Reply:
Max age tells the browser how long to cache, whereas the cache decorator enables server-side caching.